app/mixins/event-wizard.js

```diff
@@ -48,28 +48,26 @@ export default Mixin.create(MutableArray, CustomFormMixin, {
   async saveEventData(propsToSave = []) {
     const event = this.get('model.event');
     const data = {};
-    for (const property of propsToSave) {
+    await Promise.all(propsToSave.map(property => {
       try {
-        data[property] = await event.get(property);
+        return data[property] = event.get(property);
```

Comment:
Wrong

Reply:
Then how should I return the ```data[property]``` value to await on ```Promise.all()```?

Reply:
You have to iterate over the result of Promise.all

It can be also done using reduce in less lines of code but it'll be unreadable